### PR TITLE
make update feature generic so it can be reused

### PR DESCRIPTION
**Pull request by rawlingsj**
@jimmidyson not sure if this is the best approach but; I've vendored the minishift update.go package in gofabric8 and reused the `MaybeUpdate`  function which works great now that gofabric8 releases are in the same form as minikube and minishift!  - by "the same form" I mean they have a checksum and are not tarballs.

I've also tested `update.MaybeUpdateFromGithub(os.Stdout)` which still works to update minishift as the default.

I had to make these small changes so that the update function is generic enough for both gofabric8 and minishift.

I wasn't sure if vendoring this package is the correct way or if we should split out the update into its  own repo but figured I'd start here as it was easier and it works.

Let me know what you think.

### CODE REVIEW

**Comment by rawlingsj** [comment]
whoa teats!

ok so looks like the github related tests are being rate limiting. FWIW `make test` is working locally.

**Comment by jimmidyson** [comment]
That rate limiting is a problem for real users too... That's why I want to look at tuf or similar on our own servers (dedicated perhaps?). I'll look at these changes later. Thanks for the update!

**Comment by rawlingsj** [comment]
Great - thanks

File: pkg/minikube/update/update.go [hunk 1]
```diff
@@ -43,24 +43,26 @@ import (
...
-	downloadBinary          = "minishift-" + runtime.GOOS + "-" + runtime.GOARCH
 )
 
 func MaybeUpdateFromGithub(output io.Writer) {
 	MaybeUpdate(output, githubOwner, githubRepo, lastUpdateCheckFilePath)
 }
 
 func MaybeUpdate(output io.Writer, githubOwner, githubRepo, lastUpdatePath string) {
+
+	downloadBinary := githubRepo + "-" + runtime.GOOS + "-" + runtime.GOARCH
```
Comment (jimmidyson) [inline pkg/minikube/update/update.go]: Binary name will not always be githubRepo - can you make binary name configurable please.

Comment (rawlingsj) [inline pkg/minikube/update/update.go]: Sure thing

**Pull request merged**